src/converter/one/jfr/JfrReader.java

```diff
@@ -106,6 +108,13 @@ public void close() throws IOException {
         }
     }
 
+    public ByteBuffer currentChunk() {
+        ByteBuffer chunk = buf.duplicate().asReadOnlyBuffer();
+        chunk.position((int) currentChunkStart);
+        chunk.limit((int) currentChunkSize);
+        return chunk;
+    }
```

Comment:
That is not correct. `buf` is a limited window that contains a part of the recording. Chunk does not necessarily fit into the buffer.

Reply:
TODO, let's first discuss if we want to keep `ORIGINAL_PAYLOAD`. I'll fix this if we do

Reply:
Not needed anymore